### ISSUE TITLE
[ANDROID] Error while updating property (startNode|endNode)

### ISSUE THREAD

**Issue by AndreiCalazans** [IjzerenHein/react-native-shared-element#3]
## Info

Version: "react-native-shared-element": "^0.5.1"

``` 
info Fetching system and libraries information...
System:
    OS: macOS Mojave 10.14.4
    CPU: (8) x64 Intel(R) Core(TM) i7-4870HQ CPU @ 2.50GHz
    Memory: 122.52 MB / 16.00 GB
    Shell: 5.3 - /bin/zsh
  Binaries:
    Node: 10.13.0 - ~/.nvm/versions/node/v10.13.0/bin/node
    Yarn: 1.12.3 - /usr/local/bin/yarn
    npm: 6.10.2 - ~/.nvm/versions/node/v10.13.0/bin/npm
    Watchman: 4.9.0 - /usr/local/bin/watchman
  SDKs:
    iOS SDK:
      Platforms: iOS 12.4, macOS 10.14, tvOS 12.4, watchOS 5.3
    Android SDK:
      API Levels: 16, 23, 24, 25, 26, 27, 28
      Build Tools: 23.0.1, 23.0.2, 23.0.3, 24.0.1, 25.0.1, 25.0.3, 26.0.1, 26.0.2, 26.0.3, 27.0.2, 27.0.3, 28.0.0, 28.0.1, 28.0.3
      System Images: android-23 | Intel x86 Atom_64, android-23 | Google APIs Intel x86 Atom_64, android-25 | Google APIs ARM 64 v8a, android-26 | Google APIs Intel x86 Atom, android-26 | Google APIs Intel x86 Atom_64, android-26 | Google Play Intel x86 Atom, android-27 | Android TV Intel x86 Atom, android-27 | Google Play Intel x86 Atom, chromeos-67 | Chrome OS 67
      Android NDK: 18.1.5063045
  IDEs:
    Android Studio: 3.2 AI-181.5540.7.32.5056338
    Xcode: 10.3/10G8 - /usr/bin/xcodebuild
  npmPackages:
    react: 16.8.1 => 16.8.1 
    react-native: 0.61.2 => 0.61.2

```

## Bug

When running on android it crashes with the following:

![Screen Shot 2019-10-08 at 11 49 45](https://user-images.githubusercontent.com/20777666/66407131-0f885c00-e9c3-11e9-9346-f529d5724674.png)

``` 
10-08 11:49:33.699  4070  4070 E unknown:ViewManager: Error while updating prop startNode
10-08 11:49:33.699  4070  4070 E unknown:ViewManager: java.lang.reflect.InvocationTargetException
10-08 11:49:33.699  4070  4070 E unknown:ViewManager: 	at java.lang.reflect.Method.invoke(Native Method)
10-08 11:49:33.699  4070  4070 E unknown:ViewManager: 	at com.facebook.react.uimanager.ViewManagersPropertyCache$PropSetter.updateViewProp(ViewManagersPropertyCache.java:83)
10-08 11:49:33.699  4070  4070 E unknown:ViewManager: 	at com.facebook.react.uimanager.ViewManagerPropertyUpdater$FallbackViewManagerSetter.setProperty(ViewManagerPropertyUpdater.java:134)
10-08 11:49:33.699  4070  4070 E unknown:ViewManager: 	at com.facebook.react.uimanager.ViewManagerPropertyUpdater.updateProps(ViewManagerPropertyUpdater.java:54)
10-08 11:49:33.699  4070  4070 E unknown:ViewManager: 	at com.facebook.react.uimanager.ViewManager.updateProperties(ViewManager.java:48)
10-08 11:49:33.699  4070  4070 E unknown:ViewManager: 	at com.facebook.react.uimanager.NativeViewHierarchyManager.createView(NativeViewHierarchyManager.java:278)
10-08 11:49:33.699  4070  4070 E unknown:ViewManager: 	at com.facebook.react.uimanager.UIViewOperationQueue$CreateViewOperation.execute(UIViewOperationQueue.java:179)
10-08 11:49:33.699  4070  4070 E unknown:ViewManager: 	at com.facebook.react.uimanager.UIViewOperationQueue$DispatchUIFrameCallback.dispatchPendingNonBatchedOperations(UIViewOperationQueue.java:972)
10-08 11:49:33.699  4070  4070 E unknown:ViewManager: 	at com.facebook.react.uimanager.UIViewOperationQueue$DispatchUIFrameCallback.doFrameGuarded(UIViewOperationQueue.java:943)
10-08 11:49:33.699  4070  4070 E unknown:ViewManager: 	at com.facebook.react.uimanager.GuardedFrameCallback.doFrame(GuardedFrameCallback.java:28)
10-08 11:49:33.699  4070  4070 E unknown:ViewManager: 	at com.facebook.react.modules.core.ReactChoreographer$ReactChoreographerDispatcher.doFrame(ReactChoreographer.java:174)
10-08 11:49:33.699  4070  4070 E unknown:ViewManager: 	at com.facebook.react.modules.core.ChoreographerCompat$FrameCallback$1.doFrame(ChoreographerCompat.java:84)
10-08 11:49:33.699  4070  4070 E unknown:ViewManager: 	at android.view.Choreographer$CallbackRecord.run(Choreographer.java:909)
10-08 11:49:33.699  4070  4070 E unknown:ViewManager: 	at android.view.Choreographer.doCallbacks(Choreographer.java:723)
10-08 11:49:33.699  4070  4070 E unknown:ViewManager: 	at android.view.Choreographer.doFrame(Choreographer.java:655)
10-08 11:49:33.699  4070  4070 E unknown:ViewManager: 	at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:897)
10-08 11:49:33.699  4070  4070 E unknown:ViewManager: 	at android.os.Handler.handleCallback(Handler.java:789)
10-08 11:49:33.699  4070  4070 E unknown:ViewManager: 	at android.os.Handler.dispatchMessage(Handler.java:98)
10-08 11:49:33.699  4070  4070 E unknown:ViewManager: 	at android.os.Looper.loop(Looper.java:164)
10-08 11:49:33.699  4070  4070 E unknown:ViewManager: 	at android.app.ActivityThread.main(ActivityThread.java:6541)
10-08 11:49:33.699  4070  4070 E unknown:ViewManager: 	at java.lang.reflect.Method.invoke(Native Method)
10-08 11:49:33.699  4070  4070 E unknown:ViewManager: 	at com.android.internal.os.Zygote$MethodAndArgsCaller.run(Zygote.java:240)
10-08 11:49:33.699  4070  4070 E unknown:ViewManager: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:767)

```
Full android log:
[android_log.txt](https://github.com/IjzerenHein/react-native-shared-element/files/3703121/android_log.txt)

## Reproduction

To reproduce I made a simple repro, the code works as expected on iOS. 

See this repro: https://github.com/AndreiCalazans/rnshared-android-repro

Here is a loom video as well displaying the bug. 

https://www.loom.com/share/f64c878383e348779f407f7849923d99

**PS**

Thanks for the library, great work.

**Comment by IjzerenHein** [comment]
Hi! Thanks for the extensive bug report and repro repo!
I've created a PR on the repo which addresses the problems. The main cause of the problem was that an Android the ancestor View was optimized away in the View-hierarchy. Adding `collapsable=false` fixed the problem.
I then noticed another problem, and for some reason the `Animated.timing` function required the use of `useNativeDriver`. I'm not sure why, but no position changes were registered in the native code without it. Also, the `Animated.Value` was reset to `0` after running the timing function. It shouldn't do either of those things and I think it's a react-native bug. I explicitly added `position.setValue` to address that last problem.

cheers, Hein
